File: MAP/Laborator3/src/validator/ValidatorStudent.java
package validator;

import domain.Student;
import repository.ValidationException;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * Clasa Validator Student
 * Validarea datelor de intrare pentru student
 */
public class ValidatorStudent implements Validator<Student> {
    private static Pattern usrNamePtrn = Pattern.compile("^[A-Za-z ,.'-]+$");
    private static Pattern usrIdPtrn = Pattern.compile("^[1-9][0-9]{1,}$");
    private static Pattern usrGrupaPtrn = Pattern.compile("^[1-9]{3}$");
    private static Pattern usrEmailPtrn = Pattern.compile("^[a-z0-9._%+-]+@[a-z0-9.-]+\\.[a-z]{2,4}$");

    /**
     * Validarea numelui studentului
     * @param name - String (contine litere mari/mici si caracterele speciale: ,.'-)
     * @throws ValidationException daca numele nu e valid
     */
    private static void validateName(String name) {
        Matcher mtch = usrNamePtrn.matcher(name);
        if(!mtch.matches()){
            throw new ValidationException("Nume incorect!");
        }
    }

    /**
     * Validarea id-ului studentului
     * @param id - numar intreg >0
     * @throws ValidationException daca id-ul nu e valid
     */
    private static void validateID(String id){
        Matcher mtch = usrIdPtrn.matcher(id);
        if(!mtch.matches()){
            throw new ValidationException("Id incorect!");
        }
    }

    /**
     * Validarea id-ului studentului
     * @param grupa - numar intreg de 3 cifre
     * @throws ValidationException daca id-ul nu e valid
     */
    private static void validateGrupa(String grupa) {
        Matcher mtch = usrGrupaPtrn.matcher(grupa);
        if(!mtch.matches()){
            throw new ValidationException("Grupa incorecta!");
        }
    }

    /**
     * Validarea email-ului studentului
     * @param email - String
     * @throws ValidationException daca email-ul nu e valid
     */
    private static void validateEmail(String email)  {
        Matcher mtch = usrEmailPtrn.matcher(email);
        if(!mtch.matches()){
            throw new ValidationException("Email incorect!");
        }
    }

    /**
     * Validarea entitatii Student
     * @param entity - entitatea de validat
     */
    @Override
    public void validate(Student entity)  {
        validateID(entity.getID());
        validateName(entity.getNume());
        validateGrupa(entity.getGrupa());
        validateEmail(entity.getEmail());
        validateName(entity.getIndrumatorLab());
    }
}

File: MAP/Laborator11-14/CatalogMAP/CatalogMAP/validator/ValidatorStudent.cs
using CatalogMAP.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace CatalogMAP.validator
{
    public class ValidatorStudent : IValidator<Student>
    {
        private void ValidareNume(string name)
        {
            Match match = Regex.Match(name, @"^[A-Za-z ,.'-]+$");
            if (!match.Success)
                throw new ValidationException("Nume incorect!");
        }

        private void ValidareID(string id)
        {
            Match match = Regex.Match(id, @"^[1-9][0-9]{1,}$");
            if (!match.Success)
                throw new ValidationException("ID incorect!");
        }

        private void ValidareGrupa(string grupa)
        {
            Match match = Regex.Match(grupa, @"^[1-9]{3}$");
            if (!match.Success)
                throw new ValidationException("Grupa incorecta!");
        }

        private void ValidareEmail(string email)
        {
            Match match = Regex.Match(email, @"^([\w-\.]+)@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.)|(([\w-]+\.)+))([a-zA-Z]{2,4}|[0-9]{1,3})(\]?)$");
            if (!match.Success)
                throw new ValidationException("Email incorect!");
        }

        public void Validate(Student entity)
        {
            ValidareNume(entity.Nume);
            ValidareID(entity.ID);
            ValidareGrupa(entity.Grupa);
            ValidareEmail(entity.Email);
            ValidareNume(entity.IndrumatorLab);
        }
    }
}
